src/Microsoft.DotNet.Build.Tasks/VersionTools/BaseDependenciesTask.cs

```diff
@@ -240,6 +254,16 @@ private FileRegexUpdater CreateXmlUpdater(ITaskItem step)
             return updater;
         }
 
+        private FileUpdater ConfigureFileUpdater(FileUpdater updater, ITaskItem step)
```

Comment:
I'm curious why FileRegexUpdater doesn't inherit from FileUpdater....

Reply:
Yeah, structure isn't great there. `FileRegexUpdater` came first, and it never got a significant refactor. For this and the last PR I touched these I didn't want to take up the effort to avoid breaking something. (No automated testing to speak of.)